### PR TITLE
Two factor otp

### PR DESCRIPTION
**Pull request by emord**
two factor authentication feature flag

Uses django-two-factor-auth. This only uses OTP (Google Auth) but can easily(only one line + a couple of localsettings) use twilio for call/sms as well: http://django-two-factor-auth.readthedocs.org/en/1.2.0/configuration.html#twilio-gateway

old screenshots, but general idea here:
https://docs.google.com/document/d/1mpAO2TtPt4vOqqDuRWDMCyV3FVYFGcQ1SJfSvpA_R8Q/edit

@czue 

cc: @TylerSheffels

### CODE REVIEW

File: settings.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 import os
 from urllib import urlencode
 from django.contrib import messages
+from django.core.urlresolvers import reverse_lazy
```
Comment (dimagimon) [inline settings.py]: F401 'reverse_lazy' imported but unused

File: settings.py [hunk 2]
```diff
@@ -192,7 +194,10 @@
     "corehq.util.context_processors.raven",
 ]
 
-TEMPLATE_DIRS = []
+location = lambda x: os.path.join(os.path.dirname(os.path.realpath(__file__)), x)
+TEMPLATE_DIRS =(
```
Comment (dimagimon) [inline settings.py]: E225 missing whitespace around operator

**Comment by czue** [comment]
this is boss

File: corehq/apps/hqwebapp/views.py [hunk 3]
```diff
@@ -378,6 +385,29 @@ def domain_login(req, domain, template_name="login_and_password/login.html"):
...
+    extra_context = {}
+
+    def get_context_data(self, **kwargs):
+        context = super(HQLoginView, self).get_context_data(**kwargs)
+        context.update(self.extra_context)
+        return context
+
+
+
+class CloudCareLoginView(HQLoginView):
```
Comment (dimagimon) [inline corehq/apps/hqwebapp/views.py]: E303 too many blank lines (3)

**Comment by emord** [comment]
superseded by #10010

**Pull request closed without merging**